core/execution/spot/market.go

```diff
@@ -0,0 +1,2875 @@
+		fee := num.UintZero()
+		if fees != nil {
+			fee = fees.TotalFeesAmountPerParty()[trade.Buyer]
+		}
+
+		// release buyer's trade + fees quote quantity from the holding account
+		transfer, err := m.orderHoldingTracker.ReleaseQuantityHoldingAccount(ctx, trade.BuyOrder, trade.Buyer, m.quoteAsset, scaleQuoteQuantityToAssetDP(trade.Size, trade.Price, m.positionFactor), fee)
+		if err != nil {
+			m.log.Panic("failed to release funds from holding account for trade", logging.Trade(trade))
+		}
+		transfers = append(transfers, transfer)
+
+		// release seller's base quantity from the holding account
+		transfer, err = m.orderHoldingTracker.ReleaseQuantityHoldingAccount(ctx, trade.SellOrder, trade.Seller, m.baseAsset, scaleBaseQuantityToAssetDP(trade.Size, m.baseFactor), num.UintZero())
+		if err != nil {
+			m.log.Panic("failed to release funds from holding account for trade", logging.Trade(trade))
+		}
+		transfers = append(transfers, transfer)
+	} else {
+		// if there is an aggressor, then we need to release the passive side from the holding account
+		if trade.Aggressor == types.SideSell { // the aggressor is the seller so we need to release funds for the buyer from holding
+			transfer, err := m.orderHoldingTracker.ReleaseQuantityHoldingAccount(ctx, trade.BuyOrder, trade.Buyer, m.quoteAsset, scaleQuoteQuantityToAssetDP(trade.Size, trade.Price, m.positionFactor), num.UintZero())
+			if err != nil {
+				m.log.Panic("failed to release funds from holding account for trade", logging.Trade(trade))
+			}
+			transfers = append(transfers, transfer)
+		} else { // the aggressor is the buyer, we release the funds for the seller from holding account
+			transfer, err := m.orderHoldingTracker.ReleaseQuantityHoldingAccount(ctx, trade.SellOrder, trade.Seller, m.baseAsset, scaleBaseQuantityToAssetDP(trade.Size, m.baseFactor), num.UintZero())
+			if err != nil {
+				m.log.Panic("failed to release funds from holding account for trade", logging.Trade(trade))
+			}
+			transfers = append(transfers, transfer)
+		}
+	}
+
+	// transfer base to buyer
+	transfer, err := m.collateral.TransferSpot(ctx, trade.Seller, trade.Buyer, m.baseAsset, scaleBaseQuantityToAssetDP(trade.Size, m.baseFactor))
+	if err != nil {
+		m.log.Panic("failed to complete spot transfer", logging.Trade(trade))
+	}
+	transfers = append(transfers, transfer)
+	sellerPayoutAmt := scaleQuoteQuantityToAssetDP(trade.Size, trade.Price, m.positionFactor)
+	if fees != nil {
+		sellerFees, ok := fees.TotalFeesAmountPerParty()[trade.Seller]
+		if ok && !sellerFees.IsZero() {
+			sellerPayoutAmt.Sub(sellerPayoutAmt, sellerFees)
+		}
+	}
+	// transfer quote (potentially minus fees) to the seller
+	transfer, err = m.collateral.TransferSpot(ctx, trade.Buyer, trade.Seller, m.quoteAsset, sellerPayoutAmt)
+	if err != nil {
+		m.log.Panic("failed to complete spot transfer", logging.Trade(trade))
+	}
+	transfers = append(transfers, transfer)
+
+	// now pay fees - always from the buyer
```

Comment:
I find this comment quite confusing, it's the aggressive party that pays the fees, not necessary the buyer. 
I get that there are difference between buyer and seller todo with base and quote asset, but that should've falsify the statement I made above.

Fees are the most confusing part of this PR for me, it looks from the tests that everything is fine and they are implemented correctly, but I struggle to convince myself of that from just reading the code alone.

Reply:
ok, do you understand the idea though? the aggressor always covers the fee, but this is about who it's taken from. If the aggressor is the buyer, then the fee is additive to the cost of the trade, i.e. buyer pays (in quote) the cost of the trade + fee. If the aggressor is a seller, then they still pay the fee, but by way of getting less of the trade's value so the fees are taken from what the buyer is supposed to pay the seller.

Reply:
Yes I get the idea, but it's just very difficult for me to work it out from the code. What you explained above would make for a good comment in the code I think.